app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,18 @@
+class CommentsController < ApplicationController
+
+    def create
+        @comment = Comment.new(comment_params)
+        if @comment.save! 
+            flash[:success] = "New comment added!"
+            redirect_back(fallback_location: root_path) 
+        else
+            flash[:error] = "Failed to add a new comment."
+            redirect_back(fallback_location: root_path)
+        end
+    end
+
+    private
+    def comment_params
+        params.require(:comment).permit(:comment_text, :post_id, :user_id)
```

Comment:
Have a look at [nested resources](http://guides.rubyonrails.org/routing.html#nested-resources) -- it will help to prevent injection in situations like this since you won't be passing blog and user ids as parameters directly to the comment. Comments will always be on a post, so you can do something like this:
```ruby
class CommentsController < ApplicationController
  before_action :set_post
  # ...
  def create
    @comment = @post.comments.build(comment_params)
    @comment.user = current_user
    if @comment.save
      # ...
    else
      # ...
    end
  end
  # ...
  private
  def comment_params
    params.require(:comment).permit(:comment_text)
  end
  
  def set_post
    @post = Post.find(params[:post_id])
  end
end
```
This would require the routes to be nested:
```ruby
# routes.rb
resources :posts do
  resources :comments
end
```
You can nest multiple levels deep, too, so you could have comments within posts within blogs within users if you wanted, although 2-3 is usually the maximum recommended.

Reply:
Will do, thanks for the link. I did realise I was essentially hacking the solution together at this point, but I hadn't come across the nested resources feature yet.